### PR TITLE
hab bldr job status

### PR DESCRIPTION
**Pull request by raskchanky**
This implements the `hab bldr job status` command, which can take either a group id or `--origin <origin>`.  When given a group id, it will retrieve the status from jobsrv for that job group and pretty print it.  There is also a new API endpoint which will retrieve the status for all job groups for an origin.  This sounded useful when I thought it up, and now I'm less sure of its utility, but it's there anyway.

tenor-38674633

Closes https://github.com/habitat-sh/habitat/issues/3959

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-jobsrv/src/data_store.rs [hunk 1]
```diff
@@ -603,15 +630,7 @@ impl DataStore {
...
-        let group_state = match &js[..] {
-            "Dispatching" => jobsrv::JobGroupState::GroupDispatching,
-            "Pending" => jobsrv::JobGroupState::GroupPending,
-            "Complete" => jobsrv::JobGroupState::GroupComplete,
-            "Failed" => jobsrv::JobGroupState::GroupFailed,
-            "Queued" => jobsrv::JobGroupState::GroupQueued,
-            "Canceled" => jobsrv::JobGroupState::GroupCanceled,
-            _ => return Err(Error::UnknownJobGroupState),
-        };
+        let group_state = js.parse::<jobsrv::JobGroupState>()?;
```
Comment (chefsalim) [inline components/builder-jobsrv/src/data_store.rs]: Good clean-up!

File: components/builder-protocol/src/jobsrv.rs [hunk 2]
```diff
@@ -426,11 +441,42 @@ impl Serialize for JobGroupState {
             2 => serializer.serialize_str("Complete"),
             3 => serializer.serialize_str("Failed"),
             4 => serializer.serialize_str("Queued"),
+            5 => serializer.serialize_str("Canceled"),
```
Comment (chefsalim) [inline components/builder-protocol/src/jobsrv.rs]: Another good catch!

**Comment by chefsalim** [comment]
tenor-46317558

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/298635806).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by raskchanky** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
Sorry; I had a problem testing this pull request. The error was:

```ruby
Expected process to exit with [0], but received '128'
---- Begin output of git fetch origin pull/4003/merge ----
STDOUT: 
STDERR: Warning: Permanently added 'github.com,192.30.255.113' (RSA) to the list of known hosts.
fatal: Couldn't find remote ref pull/4003/merge
---- End output of git fetch origin pull/4003/merge ----
Ran git fetch origin pull/4003/merge returned 128
```

Oops

**Pull request merged**